numpy/testing/utils.py

```diff
@@ -689,20 +689,13 @@ def chk_same_position(x_id, y_id, hasval='nan'):
 
         if isnumber(x) and isnumber(y):
             x_isnan, y_isnan = isnan(x), isnan(y)
-            x_isinf, y_isinf = isinf(x), isinf(y)
```

Comment:
Previous logic for `assert_array_compare` was if there was `inf` in `x` or `y`, it should be at the same location (array id). It just checks the location of `inf`, it does not do any sort of comparison, which is wrong.